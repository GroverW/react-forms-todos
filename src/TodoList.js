import React, { useState } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import uuid from 'uuid/v4';

const TodoList = () => {
  const [ todos, setTodos ] = useState([]);

  const addTodo = (task) => {
    let newTask = {task, id:uuid()};
    setTodos(todos => [...todos, newTask]);
  }

  const removeTodo = (id) => {
    const newList = todos.filter(todo => todo.id !== id);
    setTodos(newList);
  }

  const editTodo = ({id, task}) => {
    let newList = todos.map(todo => id === todo.id ? {...todo, task:task} : todo);
    setTodos(newList);
  }

  return (
    <div className="TodoList">
      <NewTodoForm addTodo={addTodo}/>
      {todos.map(todo => (
        <Todo id={todo.id} task={todo.task} removeTodo={removeTodo} editTodo={editTodo} key={todo.id}/>
      ))}
    </div>
  );
};

export default TodoList;