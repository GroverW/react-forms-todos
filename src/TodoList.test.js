import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import TodoList from './TodoList';


it('renders without crashing', () => {
  render(<TodoList />);
});

it('matches the snapshot', () => {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it('successfully adds task to list', () => {
  const { queryByText, queryByLabelText } = render(<TodoList />);

  createTask(queryByLabelText, queryByText);

  expect(queryByText("Test Task")).toBeInTheDocument();
  expect(queryByText("Delete")).toBeInTheDocument();
});

it('successfully removes task from list', () => {
  const { queryByText, queryByLabelText } = render(<TodoList />);

  createTask(queryByLabelText, queryByText);

  expect(queryByText("Test Task")).toBeInTheDocument();
  expect(queryByText("Delete")).toBeInTheDocument();

  const deleteBtn = queryByText("Delete");

  fireEvent.click(deleteBtn);
  expect(queryByText("Test Task")).not.toBeInTheDocument();
  expect(queryByText("Delete")).not.toBeInTheDocument();

});

function createTask(queryByLabelText, queryByText) {
  const taskInput = queryByLabelText("New Task:");
  const submitBtn = queryByText("Add!");

  fireEvent.change(taskInput, {target: {value: "Test Task"}});
  fireEvent.click(submitBtn);
}