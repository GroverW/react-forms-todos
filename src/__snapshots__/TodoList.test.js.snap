// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form>
      <label
        for="task"
      >
        New Task:
      </label>
      <input
        id="task"
        name="newTask"
        type="text"
        value=""
      />
      <button>
        Add!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
